tvx/chain/types/message.go

```diff
@@ -0,0 +1,350 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	crypto "github.com/filecoin-project/specs-actors/actors/crypto"
+	block "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+type Message struct {
```

Comment:
do we need to import this? can we reference the `lotus/chain/types` variant directly?

Reply:
I am not sure - there are different fields (for example `CallSeqNum`, and `Version`), and there is a test that we should migrate which is testing `CallSeqNum`.

Reply:
Actually you are right about that. `exec-lotus` is trying to deserialise the `types.Messages`, so this is what the test vector should marshal. Will update and see if we can still keep some of the tests for the nonce applicable.

Reply:
Added as a TODO. Will address.